pafy/backend_shared.py

```diff
@@ -400,6 +400,28 @@ def _sortkey(x, keybitrate=0, keyftype=0):
         else:
             return r
 
+    @classmethod
+    def _content_available(cls, url):
+        try:
+            response = urlopen(url)
+        except HTTPError:
+            return False
+        else:
+            return response.getcode() < 300
+
+    def getbestthumb(self):
+        part_url = "http://i.ytimg.com/vi/%s/" % self.videoid
+        # Thumbnail resolution sorted in descending order
+        thumbs = ("maxresdefault.jpg",
+                  "sddefault.jpg",
+                  "hqdefault.jpg",
+                  "mqdefault.jpg",
+                  "default.jpg")
+        for thumb in thumbs:
+            url = part_url + thumb
+            if self._content_available(url):
```

Comment:
I wonder if there's a better way to do this... I don't know if there's relevant code in youtube-dl.

Reply:
You were right. Youtube-dl can do this for us. I've pushed some changes so if youtube-dl backend is set, it will fetch thumbnail URL using it otherwise fallback and test for valid thumbnail URLs.